### DISCUSSION TITLE
Supabase local development

### DISCUSSION THREAD

**Discussion by Samurai016** [supabase/supabase#13476] [Questions]
### Discussed in https://github.com/orgs/supabase/discussions/10975

<div type='discussions-op-text'>

<sup>Originally posted by **ANYURU** December 14, 2022</sup>
I have started a new project on supabase and I have some challenges setting up the supabase environment for local development.
Whenever I try to connect the local database to the remote database, I get an error about the supabase local as described below.
On initialising a directory into a supabase project by default the major version is set up to 14. Later it screams that it is incompatible with my remote version when I try to connect the local project to the remote project. I'd like some help in regards to how I can go about this. Thanks. 
```bash
supabase link --project-ref <project ref>
```
I get the output below
```bash
Error: Failed reading config: Invalid db.major_version: 15.
```

</div>

**Comment by silentworks** [comment]
This is due to the version in one of your systems being different. Search your `supabase/config.toml` for `major_version` and see what number it's equal to and then in your remote database open the SQL editor and run this query to get the version number.

```sql
select * from
  (select version()) as version,
  (select current_setting('server_version_num')) as version_number;
```

If these numbers are not matching then you will need to either update the version in the `supabase/config.toml` or upgrade your remote database to version 15.

**Comment by Samurai016** [comment]
Since this issue has been converted into a discussion, I close it as a duplicate and continue on the original discussion.
